File: benchmarks/cross-language/sorting/bench.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static List<long> Quicksort(List<long> arr)
    {
        if (arr.Count <= 1) return arr;
        long pivot = arr[arr.Count / 2];
        var left = arr.Where(x => x < pivot).ToList();
        var middle = arr.Where(x => x == pivot).ToList();
        var right = arr.Where(x => x > pivot).ToList();
        var result = Quicksort(left);
        result.AddRange(middle);
        result.AddRange(Quicksort(right));
        return result;
    }

    static void Main()
    {
        var rng = new Random(42);
        for (int i = 0; i < 100; i++)
        {
            var data = Enumerable.Range(0, 1000).Select(_ => (long)rng.Next(10001)).ToList();
            var sorted = Quicksort(data);
        }
        Console.WriteLine("sorted 100 arrays");
    }
}

File: benchmarks/cross-language/sorting/bench.py
import random

def quicksort(arr: list[int]) -> list[int]:
    if len(arr) <= 1:
        return arr
    pivot = arr[len(arr) // 2]
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]
    return quicksort(left) + middle + quicksort(right)

def main():
    random.seed(42)
    for i in range(100):
        data = [random.randint(0, 10000) for _ in range(1000)]
        sorted_data = quicksort(data)
    print(f"sorted {i + 1} arrays")

if __name__ == "__main__":
    main()
